src-java/nbworker-topology/nbworker-storm-topology/src/test/java/org/openkilda/wfm/topology/nbworker/services/FlowOperationsServiceTest.java

```diff
@@ -290,8 +306,98 @@ public void shouldNotUpdateVlanStatistics() throws FlowNotFoundException {
         flowOperationsService.updateFlow(new FlowCarrierImpl(), receivedFlow);
     }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void unableToSetProtectedPathForInitiallyOneSwitchFlowTest() throws FlowNotFoundException {
+        createFlow(FLOW_ID_1, switchA, switchA, false);
+        FlowPatch receivedFlow = FlowPatch.builder()
+                .flowId(FLOW_ID_1)
+                .allocateProtectedPath(true)
+                .build();
+        flowOperationsService.updateFlow(new FlowCarrierImpl(), receivedFlow);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void unableToMakeOneSwitchFlowFromProtectedByUpdatingDstTest() throws FlowNotFoundException {
+        createFlow(FLOW_ID_1, switchA, switchB, true);
+        FlowPatch receivedFlow = FlowPatch.builder()
+                .flowId(FLOW_ID_1)
+                .destination(PatchEndpoint.builder().switchId(SWITCH_ID_1).build())
+                .build();
+        flowOperationsService.updateFlow(new FlowCarrierImpl(), receivedFlow);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void unableToMakeOneSwitchFlowFromProtectedByUpdatingSrcTest() throws FlowNotFoundException {
+        createFlow(FLOW_ID_1, switchA, switchB, true);
+        FlowPatch receivedFlow = FlowPatch.builder()
+                .flowId(FLOW_ID_1)
+                .source(PatchEndpoint.builder().switchId(SWITCH_ID_2).build())
+                .build();
+        flowOperationsService.updateFlow(new FlowCarrierImpl(), receivedFlow);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void unableToMakeProtectedOneSwitchFlowTest() throws FlowNotFoundException {
+        createFlow(FLOW_ID_1, switchA, switchB, false);
+        FlowPatch receivedFlow = FlowPatch.builder()
+                .flowId(FLOW_ID_1)
+                .source(PatchEndpoint.builder().switchId(SWITCH_ID_3).build())
+                .destination(PatchEndpoint.builder().switchId(SWITCH_ID_3).build())
+                .allocateProtectedPath(true)
+                .build();
+        flowOperationsService.updateFlow(new FlowCarrierImpl(), receivedFlow);
+    }
+
+    @Test
+    public void ableToMakeProtectedFlowFromMultiSwitchFlowTest() throws FlowNotFoundException {
+        createFlow(FLOW_ID_1, switchA, switchB, false);
+        FlowPatch receivedFlow = FlowPatch.builder()
+                .flowId(FLOW_ID_1)
+                .allocateProtectedPath(true)
+                .build();
+        flowOperationsService.updateFlow(new FlowCarrierImpl(), receivedFlow);
+        // no exception expected
```

Comment:
assertj has an idiom for no exceptions thrown. There is also a way to tell the same using junit annotation `@Test(expected = None.class)`. The difference is the test fails and handled properly or the test execution crashes.

Reply:
I tried to do it, but Idea marked it as warning because (expected = None.class) is default value